### ISSUE TITLE
option --dialect doesn’t work with ClickHouse and MySQL

### ISSUE THREAD

**Issue by Rupreht** [RAprogramm/sql-query-analyzer#42]
### Issue: `--dialect` option doesn't work with ClickHouse and MySQL

**Problem description**  
When running the SQL query analyzer with the `--dialect` parameter (values `clickhouse` or `mysql`), the functionality doesn't operate correctly. The analyzer appears to ignore the specified dialect and likely falls back to the default value (`generic`).

**Steps to reproduce**  
1. Prepare SQL files for schema and queries:
   - `schema_clickhouse.sql` / `schema_mysql.sql`
   - `queries_clickhouse.sql` / `queries_mysql.sql`
2. Run the test script (see example below).
3. Verify that:
   - SQL files exist and are valid;
   - the analyzer is installed and available in PATH.

**Current execution script**  
```bash
#!/bin/bash

set -ex

WORKDIR=$(dirname $0)

# DIALECT="mysql"
DIALECT="clickhouse"
SCHEMA_FILE="$WORKDIR/schema_${DIALECT}.sql"
QUERIES_FILE="$WORKDIR/queries_${DIALECT}.sql"

if [ "${DIALECT}" == "clickhouse" ]; then
  # brew install clickhouse-sql-parser
  clickhouse-sql-parser -v

  clickhouse-sql-parser \
    -f ${SCHEMA_FILE} \
    -format 1>/dev/null

  clickhouse-sql-parser \
    -f ${QUERIES_FILE} \
    -format 1>/dev/null
fi

sql-query-analyzer analyze \
  --verbose --dry-run \
  --dialect ${DIALECT:-generic} \
  --output-format yaml \
  --schema ${SCHEMA_FILE:-$WORKDIR/schema.sql} \
  --queries ${QUERIES_FILE:-$WORKDIR/queries.sql}
```

**Expected behavior**  
The analyzer should:
- respect syntax and semantic specifics of the specified dialect (`clickhouse` / `mysql`);
- properly handle constructs specific to the selected dialect;
- output analysis results in YAML format according to the dialect.

**Actual behavior**  
- The analyzer behaves as if no dialect was specified (defaults to `generic`).
- ClickHouse/MySQL‑specific constructs may be processed with errors or ignored.
- There are no indications in the output that the dialect was applied.

**Additional information**  
- SQL files are attached in an archive (filenames follow the pattern `schema_<dialect>.sql` and `queries_<dialect>.sql`).
- Analyzer version: [specify if known].
- OS: [specify, e.g., macOS/Linux].
- ClickHouse parser version: [specify if known].

**Suggested fixes**  
1. Verify that the `--dialect` value is correctly passed to the analyzer's code.
2. Ensure that supported dialects are explicitly listed in the documentation/code.
3. Add logging of the current dialect within the analyzer for debugging purposes.
4. Test with other dialects (e.g., `postgresql`) to rule out the issue being specific to ClickHouse/MySQL.

[tests.tar.gz](https://github.com/user-attachments/files/23838811/tests.tar.gz)

**Comment by Rupreht** [comment]
In test.sh `set -ex` option -e - Exit immediately, if a pipeline exits with a non-zero status.

```
brew install clickhouse-sql-parser
bash ./tests/test.sh
...
+ clickhouse-sql-parser -f ./tests/schema_clickhouse.sql -format
+ clickhouse-sql-parser -f ./tests/queries_clickhouse.sql -format
+ sql-query-analyzer analyze --verbose --dry-run --dialect clickhouse --output-format yaml --schema ./tests/schema_clickhouse.sql --queries ./tests/queries_clickhouse.sql
Error: Bad request
```

**Comment by Rupreht** [comment]
@RAprogramm Hi! So, I used **DBeaver** (that's a database tool, in case you're not familiar) to create this SQL file called `schema_clickhouse.sql`. It's basically a blueprint for our ClickHouse database. After creating the file, I ran a quick check to make sure everything was in order. I used this command:
```sh
# brew install clickhouse-sql-parser
clickhouse-sql-parser -f ./tests/schema_clickhouse.sql
```
No errors, everything looks good to go.

If you have any questions or need further clarification, do not hesitate to contact me.

Best regards.
